shell/platform/linux/fl_text_input_plugin.cc

```diff
@@ -290,17 +292,27 @@ static void im_commit_cb(FlTextInputPlugin* self, const gchar* text) {
   FlTextInputPluginPrivate* priv = static_cast<FlTextInputPluginPrivate*>(
       fl_text_input_plugin_get_instance_private(self));
   std::string text_before_change = priv->text_model->GetText();
+  flutter::TextRange composing_before_change =
+      priv->text_model->composing_range();
   flutter::TextRange selection_before_change = priv->text_model->selection();
+  gboolean wasComposing = priv->text_model->composing();
 
   priv->text_model->AddText(text);
   if (priv->text_model->composing()) {
     priv->text_model->CommitComposing();
   }
 
   if (priv->enable_delta_model) {
-    flutter::TextEditingDelta delta = flutter::TextEditingDelta(
-        text_before_change, selection_before_change, text);
-    update_editing_state_with_delta(self, &delta);
+    flutter::TextEditingDelta* delta;
+    if (wasComposing) {
+      delta = new flutter::TextEditingDelta(text_before_change,
+                                            composing_before_change, text);
```

Comment:
Avoid new/delete in C++ code. Use `std::unique_ptr<Foo>`, which you can create with `std::make_unique<Foo>(constructor, args, go, here)`;

Reply:
Generally always use RAII-based smart pointers (primarily unique_ptr) unless it's impossible, they're guaranteed to clean themselves up at end of scope whether it's a return or throw, and they're effectively zero overhead.

In this case there's only one way out of the scope, but there's a function call in there that could do something crazy like throw (but almost certainly doesn't 😉). The more ways out of a scope the more `delete`s you'll need to make sure not to forget. Smart pointers guarantee cleanup so why even risk it?

In modern c++ new/delete are generally considered bad practice unless absolutely necessary. In our codebase there are maybe two spots I've found where we couldn't use either unique_ptr or shared_ptr, and we still have a new/delete, but we should avoid adding any more.

More reading in [the C++ FAQ](https://isocpp.org/wiki/faq/freestore-mgmt#memory-leaks) and [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-resource).